### PR TITLE
Add a Contributing file

### PR DESCRIPTION
**Pull request by Flaburgan**
As [suggested on discourse](https://discourse.diasporafoundation.org/t/learn-to-code-review/677/6)

Pinging @goobertron for the english verifications ;)

### CODE REVIEW

**Comment by goobertron** [comment]
Oh, thank you Fla for doing this! I had been meaning to, but am a bit shaky on the technical side. I'll have a proper look at the English tomorrow (just a couple of tiny changes, I think).

**Review by SuperTux88 — APPROVED** [review]
The technical side looks good, but I am a bit shaky on the english side ;)

**Comment by goobertron** [comment]
Here are my suggestions: https://github.com/goobertron/diaspora/commit/bba75b761823d4f4e86ee7f0e83e2acb3a7b8d26

Basically:
- a little bit of clean-up to the English,
- a sentence about what to do if your language is not on WebTranslateIt,
- add a link to /Getting_started_with_contributing on the wiki.

Hope that helps! And thanks again for doing this.

File: CONTRIBUTING.md
Comment (Flaburgan) [inline CONTRIBUTING.md]: >If you want to contribute to a language that is not featured on WebTranslateIt, please contact our developers to find out how to proceed

Not too sure about that, I'd like to redirect to the wiki page, but we definitely need to update its content. @jhass you were the one who managed translations, what's the current recommendations to contribute to translations? Do you have some time to update https://wiki.diasporafoundation.org/Contribute_translations ?

**Comment by Flaburgan** [comment]
Thank you for your feedback @goobertron! I integrated your commit. I will probably squash it before merging this but still, it's there :p

File: CONTRIBUTING.md
Comment (goobertron) [inline CONTRIBUTING.md]: I added that because I \*think\* that is how things have been done in the past for a new language. But I'm by no means sure. I suppose it would have been better if I'd have asked Jonne and @denschub rather than just adding that sentence to my PR! Hopefully they can confirm for you what is the proper procedure.

Comment (Flaburgan) [inline CONTRIBUTING.md]: We currently have to tell contributors (who want to add a new language) what to do because the wiki page is outdated. Instead of telling people come on IRC, hoping that someone who knows will be available, we should write his answer on the wiki.

Comment (denschub) [inline CONTRIBUTING.md]: The list of "Languages handled outside of WebTranslateIt" in the wiki is quite accurate.

Comment (goobertron) [inline CONTRIBUTING.md]: I was thinking also about how to handle a language that is not yet part of D* at all. I don't think we have any instructions for that. Let's add a section to the wiki article about what to do for a completely new language, and then the final sentence in the snippet above can be changed to:

> If you want to contribute to a language that is not featured on WebTranslateIt, please read the wiki article above to find out how to proceed.

**Comment by Flaburgan** [comment]
Alright, corrected, and commits squashed. Should be ready to go and I'll dig into the wiki page.

**Comment by Flaburgan** [comment]
I reorganized https://wiki.diasporafoundation.org/Contribute_translations please review. It feels way clearer for me now.

**Comment by goobertron** [comment]
That looks a lot better, thanks. I don't fully understand how this works, so excuse me if this is wrong, but in the paragraph where you say 'Copy the en files (listed below), choose the correct language code (see above) and rename every occurrence of "en" in the files names and in the code.' – while this is often correct for an entirely new language, isn't it best to copy the files of the locale on which your new language is based, and to which strings will fall-back if there is no translation specified in the new locale? E.g. for de-AT, you'd copy the de files and just leave in the strings that will be different from the standard de locale. I think!

I might now finally get around to completing a British English translation – been meaning to do this for ages.

**Comment by Flaburgan** [comment]
This is a fair point. The problem can be, we never know the state of the other languages than english (where all the keys are for sure). I don't know the opinion of others on that point, what do we want to recommend?
Also, adding a new language doesn't mean it has to be updated with github, it can also be integrated in WebTranslateIt. I need help from others, possibly @jhass to modify the page about that.
But we're going out of topic, let's focus on this PR here. There are discussion pages on the wiki if needed.

**Comment by SuperTux88** [comment]
About new languages: When you want to create a new language that is managed in WebTranslateIt (which should be the default), you don't need (and shouldn't) copy any files. A manager on WebTranslateIt needs to create the language there, and that files will be created when pulling from WebTranslateIt. You only need to manage the files by hand, when the language isn't managed with WebTranslateIt.

**Comment by SuperTux88** [comment]
@goobertron can you approve this PR when the English is OK?

**Comment by goobertron** [comment]
I've just one small change to request: to change the last sentence of line 17 to:

> If it is not already featured on WebTranslateIt, please read the wiki article above to find out how to proceed.

I think it's worth mentioning WebTranslateIt again here to make the meaning absolutely clear.

Everything else is fine.

**Comment by Flaburgan** [comment]
Done ;)

**Comment by Flaburgan** [comment]
>About new languages: When you want to create a new language that is managed in WebTranslateIt (which should be the default), you don't need (and shouldn't) copy any files. A manager on WebTranslateIt needs to create the language there, and that files will be created when pulling from WebTranslateIt. You only need to manage the files by hand, when the language isn't managed with WebTranslateIt.

But is there possible to fork this language from a based one on WebTranslateIt? I mean the de, de-AT example is accurate.

**Comment by denschub** [comment]
There is no such thing as "forking" a language. There is no need to copy an entire language. If you want to create a new language based on an existing one, you have to change the `fallbacks` array in [`local_settings.yml`](https://github.com/diaspora/diaspora/blob/d62d79bc293c89c6439ce4ec2414a17d5dc5b11e/config/locale_settings.yml#L77), that's all. Mass-duplicating segments is discouraged and bad.

**Comment by Flaburgan** [comment]
That's exactly the kind of feedback I need as I don't know how things work. Could you please update the wiki page?

**Comment by SuperTux88** [comment]
@goobertron after [this](https://discourse.diasporafoundation.org/t/i-want-to-start-contributing-to-diaspora-development-where-should-i-start/817/9) discussion on discourse, I saw that this PR is still open. Is everything OK from your side with this? We should still update the wiki, but that is out of scope of this PR. So if everything is OK, I'd like to merge this.

**Comment by goobertron** [comment]
To save me reading through the whole file again, then if nothing apart from my suggested edit has been changed since my last comment, my comment that 'Everything else is fine.' still stands.

**Comment by SuperTux88** [comment]
I don't have the commit-ID from 17 days ago anymore, so I can't check that. @Flaburgan can you confirm that you only changed the suggested edit?

**Comment by Flaburgan** [comment]
Yep, the suggestion about WebTranslateIt. Nothing else than the translation paragraph changed.

**Comment by goobertron** [comment]
Then it's still good to go, as far as I'm concerned!

**Comment by SuperTux88** [comment]
Merged, thanks @Flaburgan and @goobertron 👍

**Pull request merged**
